test/node/map.test.ts

```diff
@@ -0,0 +1,76 @@
+import { expect } from 'chai';
+import * as BSON from '../register-bson';
+
+describe('ES Map support in serialize()', () => {
+  it('serialize a empty Map to BSON document - { }', () => {
+    const map = new Map();
+    const emptyBSON = new Uint8Array([5, 0, 0, 0, 0]);
+    const result = BSON.serialize(map);
+    expect(result).to.have.property('byteLength', 5);
+    expect(result, 'byteLength must be 5 followed by null terminator').to.deep.equal(emptyBSON);
+  });
+
+  it('serialize a Map with one key to BSON document - { a: 2 }', () => {
+    const map = new Map([['a', new BSON.Int32(2)]]);
```

Comment:
per slack discussion, let's update the assertions to be easier to read by providing an annotated `bufferFromHexArray` instead of a byte by byte sequence of assertions

Reply:
done!